client/go/internal/cli/cmd/visit.go

```diff
@@ -0,0 +1,339 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+// vespa visit command
+// Author: arnej
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	"github.com/vespa-engine/vespa/client/go/internal/util"
+	"github.com/vespa-engine/vespa/client/go/internal/vespa"
+)
+
+type visitArgs struct {
+	contentCluster string
+	fieldSet       string
+	selection      string
+	makeFeed       bool
+	jsonLines      bool
+	pretty         bool
+	quietMode      bool
+	chunkCount     int
+}
+
+var totalDocCount int
+
+func newVisitCmd(cli *CLI) *cobra.Command {
+	var (
+		vArgs visitArgs
+	)
+	cmd := &cobra.Command{
+		Use:   "visit",
+		Short: "Visit all documents in a content cluster",
+		Long: `Run visiting of a content cluster to retrieve all documents.
+
+more explanation here
+even more explanation here
+`,
+		Example: `$ vespa visit # get documents from any cluster
+$ vespa visit --content-cluster search # get documents from cluster named "search"
+`,
+		Args:              cobra.MaximumNArgs(0),
+		DisableAutoGenTag: true,
+		SilenceUsage:      true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			vArgs.quietMode = cli.config.isQuiet()
+			service, err := documentService(cli)
+			if err != nil {
+				return err
+			}
+			result := probeHandler(service)
+			if result.Success {
+				result = visitClusters(vArgs, service)
+			}
+			if !result.Success {
+				return fmt.Errorf("visit failed: %s", result.Message)
+			}
+			if !vArgs.quietMode {
+				fmt.Fprintln(os.Stderr, "[debug] sum of 'documentCount':", totalDocCount)
```

Comment:
Consider adding a `printDebug` in the same style as these: https://github.com/vespa-engine/vespa/blob/master/client/go/internal/cli/cmd/root.go#L255-L271